litestar/security/session_auth/middleware.py

```diff
@@ -61,7 +60,7 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
                 exception_handlers=litestar_app.exception_handlers or {},  # pyright: ignore
                 debug=None,
             )
-            self.app = SessionMiddleware(
+            self.app = self.config.session_backend_config.middleware.middleware(
```

Comment:
Just for my understanding, why was this change needed?

Reply:
Well, before I started working on this I was experimenting with a SessionMiddleware override. But even though, I also overrode the corrsponding `DefineMiddleware` method, my custom version was not loaded. So since it is possible to place a reference to the actual session backend middleware, I thought that was the better way to do it: the session_auth should use whatever session middleware the backend provides, not just hard-code a certain middleware.